src/patterns/ec2-app/base.ts

```diff
@@ -470,6 +511,45 @@ export class GuEc2App extends Construct {
       open: access.scope === AccessScope.PUBLIC,
     });
 
+    if (props.googleAuth?.enabled) {
+      const configPrefix = `${scope.stage}/${scope.stack}/${app}`;
+      const clientIdPath = props.googleAuth.clientIdPath ?? `/${configPrefix}/googleClientID`;
+
+      const clientId = StringParameter.fromStringParameterAttributes(this, "clientID", {
+        parameterName: clientIdPath,
+      }).stringValue;
+
+      // Unfortunately, Cloudformation doesn't support directly using secret
+      // Parameter Store values. But it is possible to use Secrets Manager.
+      const secretPath = props.googleAuth.clientSecretPath ?? `${configPrefix}/clientSecret`;
+      const clientSecret = SecretValue.secretsManager(secretPath);
+
+      const authAction = ListenerAction.authenticateOidc({
```

Comment:
Should we add this listener action at the [construct level](https://github.com/guardian/cdk/tree/main/src/constructs/loadbalancing/alb) so that non-pattern users also benefit?

Reply:
Yeah, that's a nice idea. I'd hope most adoption would be via the pattern, but could see it being used standalone too.

Reply:
Having looked a bit more closely, I'm not sure this is straightforward after all. The ALB type is separate from the Listener and Target types, and the auth functionality interacts with them all, so think it is best to keep this at the pattern level for this PR at least.